com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs

```diff
@@ -566,6 +578,12 @@ static void InitializeStackedCameraData(Camera baseCamera, UniversalAdditionalCa
             int msaaSamples = 1;
             if (baseCamera.allowMSAA && settings.msaaSampleCount > 1 && rendererSupportsMSAA)
                 msaaSamples = (baseCamera.targetTexture != null) ? baseCamera.targetTexture.antiAliasing : settings.msaaSampleCount;
+#if ENABLE_VR && ENABLE_XR_MODULE
+            // Use XR's MSAA if camera is XR camera. XR MSAA needs special handle here because it is not per Camera.
+            // Multiple cameras could render into the same XR display and they should share the same MSAA level.
+            if (cameraData.xrRendering)
+                msaaSamples = XRSystem.GetMSAALevel();
```

Comment:
in the PC case, why does URP care what the XR device texture MSAA values are?  They don't need to match the camera, xr device textures should just be sample 1 and we should resolve to them as last step in camera stack.

I think on the provider layer the XR device textures are always samples 1 on PC.  Mobile is a bit different with autoresolve.

Reply:
Good point!

Reply:
The reason we want to set MSAA through XRSystem is because the MSAA is per display and we want to configure this setting on a per display basis.
Let's make an example where pipeline configures to MSAA4. 
We have 2 cases here:
1. with auto resolve: eye texture is sample 1 and implicit msaa buffer is sample 4, URP should render to implicit msaa buffer. All URP intermediate texture should be MSAA4
2. without auto resolve: eye texture is sample 1, URP should render to eye texture. All URP intermediate texture should be MSAA4 and explicit MSAA resolve pass is required to resolve from MSAA4 intermediate texture to MSAA1 eye texture.
In both cases, if camera is rendering to XR display, its intermediate textures should be configured to MSAA4.

Right now, every URP camera has XR on/off toggle. So we just check that bool, and use display's MSAA if its on. For non-XR camera, it uses the per camera MSAA setting.

Reply:
Discussed with thomas - It looks like the `XRSystem.UpdateMSAALevel` basically becomes a no-op on PC anyway and we won't go reallocating textures or anything.  The XR device texture will still always have 1 sample on PC and we resolve as expected.  And this code cleanly supports autoresolve, so I think we're good.